test/functional/specs/MediaCollection/MA2.js

```diff
@@ -0,0 +1,385 @@
+import getResponseBody from "../../helpers/networkLogger/getResponseBody";
+import createResponse from "../../helpers/createResponse";
+import { TEST_PAGE as TEST_PAGE_URL } from "../../helpers/constants/url";
+import createAlloyProxy from "../../helpers/createAlloyProxy";
+import orgMediaConfig from "../../helpers/constants/configParts/orgMediaConfig";
+import mediaCollection from "../../helpers/constants/configParts/mediaCollection";
+import { sleep } from "../Migration/helper";
+
+const networkLogger = createNetworkLogger();
+const config = compose(orgMediaConfig, mediaCollection);
+createFixture({
+  title: "Media Collection for legacy migration usecases.",
+  url: TEST_PAGE_URL,
+  requestHooks: [
+    networkLogger.edgeEndpointLogs,
+    networkLogger.mediaPauseEndpointLogs,
+    networkLogger.mediaPlayEndpointLogs,
+    networkLogger.chapterStartEndpointLogs,
+    networkLogger.pingEndpointLogs,
+    networkLogger.chapterCompleteEndpointLogs,
+    networkLogger.chapterSkipEndpointLogs,
+    networkLogger.adStartEndpointLogs,
+    networkLogger.adBreakStartEndpointLogs,
+    networkLogger.adBreakCompleteEndpointLogs,
+    networkLogger.adSkipEndpointLogs,
+    networkLogger.adCompleteEndpointLogs,
+    networkLogger.errorEndpointLogs,
+    networkLogger.sessionCompleteEndpointLogs,
+    networkLogger.sessionEndEndpointLogs,
+    networkLogger.statesUpdateEndpointLogs,
+    networkLogger.bitrateChangeEndpointLogs,
+    networkLogger.bufferStartEndpointLogs
+  ]
+});
+
+test.meta({
+  ID: "MA3",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const assertSessionStarted = async () => {
+  await responseStatus(networkLogger.edgeEndpointLogs.requests, 200);
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
+
+  const createSession = networkLogger.edgeEndpointLogs.requests[0];
+  const requestBody = JSON.parse(createSession.request.body);
+  await t.expect(requestBody.events[0].xdm.eventType).eql("media.sessionStart");
+  await t.expect(requestBody.events[0].xdm.mediaCollection.playhead).eql(0);
+  const response = JSON.parse(
+    getResponseBody(networkLogger.edgeEndpointLogs.requests[0])
+  );
+  const mediaCollectionPayload = createResponse({
+    content: response
+  }).getPayloadsByType("media-analytics:new-session");
+
+  return mediaCollectionPayload[0].sessionId;
+};
+
+const assertPingsNotSent = async () => {
+  const secondPingEventRequest = networkLogger.pingEndpointLogs.requests[2];
+  await t.expect(secondPingEventRequest).eql(undefined);
+};
+const assertPingsSent = async sessionId => {
+  const pingEventRequest = networkLogger.pingEndpointLogs.requests[0];
+  const pingEvent = JSON.parse(pingEventRequest.request.body).events[0];
+  await t.expect(pingEvent.xdm.mediaCollection.sessionID).eql(sessionId);
+  await t.expect(pingEvent.xdm.eventType).eql("media.ping");
+};
+const assertEventIsSent = async (
+  endpointLogs,
+  eventType,
+  sessionId,
+  playhead,
+  order = 0
+) => {
```

Comment:
If you add this here, then you won't need all the sleeps.
```
await t.expect(endpointLogs.count(() => true)).gte(order+1)
```
This will repeatedly check until the event is sent (with a timeout) so no need for the sleeps. It uses the Smart Query Assertion Mechanism. See https://testcafe.io/documentation/402767/reference/test-api/requestlogger/count